crossbot/models.py

```
@@ -186,6 +191,57 @@ def remove_crossword_time(self, *args, **kwargs):
     def remove_easy_sudoku_time(self, *args, **kwargs):
         return self.remove_time(EasySudokuTime, *args, **kwargs)
 
+    def add_item(self, item, amount=1):
+        """Add an item to this user's inventory.
+        Args:
+            item: An Item object.
+            amount: An integer > 0.
+        """
+        assert isinstance(item, Item)
+        assert amount > 0
+
+        record, _ = ItemOwnershipRecord.objects.get_or_create(
+            owner=self, item_key=item.key)
+        record.quantity += amount
+        record.save()
+
+    def remove_item(self, item, amount=1):
+        """Remove an item from this user's inventory.
+        Args:
+            item: The Item to remove.
+            amount: An integer > 0.
+        Returns:
+            Whether or not the item was removed.
+        """
+        assert isinstance(item, Item)
+        assert amount > 0
+
+        try:
+            record = ItemOwnershipRecord.objects.get(
+                owner=self, item_key=item.key)
+        except ItemOwnershipRecord.DoesNotExist:
+            return False
```

i think `User.removeItem` should `raise` instead of return `False` if there are insufficient things.

---

That doesn't fit with the rest of `models.py`. I've been using ret vals throughout (see: remove time etc.). It's much more common to want to gracefully handle this in other code than let the error bubble up.

---

ugh. You're right that this isn't consistent. Now that i'm thinking about it, we should raise in those cases, but w/e.